go/geth_adapter/adapter_test.go

```diff
@@ -548,6 +538,180 @@ func TestRunContextAdapter_Run(t *testing.T) {
 	}
 }
 
+func TestGethAdapter_CorruptValuesReturnErrors(t *testing.T) {
+	tests := map[string]struct {
+		firstBlock  *big.Int
+		baseFee     *big.Int
+		chainID     *big.Int
+		blobBaseFee *big.Int
+		gasPrice    *big.Int
+		difficulty  *big.Int
+	}{
+		"revision": {
+			firstBlock: big.NewInt(1000),
+		},
+		"baseFee": {
+			baseFee: big.NewInt(-1),
+		},
+		"chainID": {
+			chainID: big.NewInt(-1),
+		},
+		"blobBaseFee": {
+			blobBaseFee: big.NewInt(-1),
+		},
+		"gasPrice": {
+			gasPrice: big.NewInt(-1),
+		},
+		"difficulty": {
+			difficulty: big.NewInt(-1),
+		},
+	}
+
+	for name, test := range tests {
+		t.Run("unsupported-"+name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			stateDb := NewMockStateDb(ctrl)
+			interpreter := tosca.NewMockInterpreter(ctrl)
+
+			blockNumber := int64(42)
+			blockParameters := geth.BlockContext{
+				BlockNumber: big.NewInt(blockNumber),
+				BaseFee:     test.baseFee,
+				BlobBaseFee: test.blobBaseFee,
+				Difficulty:  test.difficulty,
+			}
+			if test.firstBlock == nil {
+				test.firstBlock = big.NewInt(1)
+			}
+			chainConfig := &params.ChainConfig{ChainID: test.chainID, IstanbulBlock: test.firstBlock}
+			evm := geth.NewEVM(blockParameters, stateDb, chainConfig, geth.Config{})
+			evm.TxContext = geth.TxContext{
+				GasPrice: test.gasPrice,
+			}
+
+			adapter := &gethInterpreterAdapter{
+				evm:         evm,
+				interpreter: interpreter,
+			}
+
+			stateDb.EXPECT().AddRefund(gomock.Any())
+
+			address := tosca.Address{0x42}
+			contract := geth.NewContract(common.Address(address), common.Address(address), nil, 0, nil)
+
+			ret, err := adapter.Run(contract, nil, false)
+			require.Error(t, err, "could not convert"+name)
+			require.Nil(t, ret, "expected nil return value")
+		})
+	}
+}
+
+func TestGethAdapter_CallForwardsToTheRightKind(t *testing.T) {
+	any := gomock.Any()
+	tests := map[string]struct {
+		kind  tosca.CallKind
+		setup func(mock *MockCallContextInterceptor)
+	}{
+		"call": {
+			kind: tosca.Call,
+			setup: func(mock *MockCallContextInterceptor) {
+				mock.EXPECT().Call(any, any, any, any, any, any).Return(
+					nil, uint64(0), nil,
+				)
+			},
+		},
+		"delegateCall": {
+			kind: tosca.DelegateCall,
+			setup: func(mock *MockCallContextInterceptor) {
+				mock.EXPECT().DelegateCall(any, any, any, any, any).Return(
```

Comment:
nit: This test might also be a good opportunity to check that the call parameters are forwarded correctly - or would this be an independent test?